src/modules/Kdump.rb

```diff
@@ -880,6 +881,117 @@ def Summary
+    #                   packages proposal is made yet
+    def free_space_for_dump_b
+      kdump_savedir = @KDUMP_SETTINGS.fetch("KDUMP_SAVEDIR", "file:///var/log/dump")
+      log.info "Using savedir #{kdump_savedir}"
+
+      if kdump_savedir.start_with?("/")
+        log.warn "Using old format"
+      elsif kdump_savedir.start_with?("file://")
+        kdump_savedir.sub!(/file:\/\//, "")
+      else
+        log.info "KDUMP_SAVEDIR #{kdump_savedir.inspect} is not local"
+        return nil
+      end
+
+      # unified format of directory
+      kdump_savedir = format_dirname(kdump_savedir)
+
+      partitions_info = SpaceCalculation.GetPartitionInfo()
+      if partitions_info.empty?
+        log.warn "No partitions info available"
+        return nil
+      end
+
+      log.info "Disk usage: #{partitions_info}"
+      # Create a hash of partitions and their free space { partition => free_space, ... }
+      # "name" usually does not start with "/", but does so for root filesystem
+      # File.join ensures that paths do not contain dulplicit "/" characters
+      partitions_info = partitions_info.map do |partition|
+        { format_dirname(partition["name"]) => partition["free"] }
+      end.inject(:merge)
+
+      # All partitions matching KDUMP_SAVEDIR
+      matching_partitions = partitions_info.select do |partition, _space|
+        kdump_savedir.start_with?(partition)
+      end
+
+      # The longest match
+      partition = matching_partitions.keys.sort_by{|partiton| partiton.length}.last
+      free_space = matching_partitions[partition]
+
+      if free_space.nil? || !free_space.is_a?(::Integer)
+        log.warn "Available space for partition #{partition} not provided (#{free_space.inspect})"
+        return nil
+      end
+
+      # packager counts in kB, we need bytes
+      free_space *= 1024
+      log.info "Available space for dump: #{free_space} bytes in #{partition} directory"
+
+      free_space
+    end
+
+    # Returns disk space in bytes requested for kernel dump (as defined in FATE#317488)
+    #
+    # @return [Integer] bytes
+    def space_requested_for_dump_b
+      # Total memory is in MB, converting to bytes
+      total_memory * 1024**2 + RESERVED_DISK_SPACE_BUFFER_B
+    end
+
+    # Returns installation proposal warning as part of the MakeProposal map result
+    # includes 'warning' and 'warning_level' keys
+    #
+    # @param returns [Hash] with warnings
+    def proposal_warning
+      return {} unless @add_crashkernel_param
+
+      free_space = free_space_for_dump_b
+      requested_space = space_requested_for_dump_b
+
+      log.info "Free: #{free_space}, requested: #{requested_space}"
+      return {} if free_space.nil? || requested_space.nil?
+
+      warning = {}
+
+      # Smaller differences are not possible to show with quite small precision
+      # in human-readable size formatting
+      if (requested_space * 0.99) < free_space && free_space < requested_space
```

Comment:
1%+ difference can be seen in the result of `String.FormatSizeWithPrecision(requested_space, 2, true)` 2 is a precision